main/src/ca/uwaterloo/flix/language/phase/Weeder.scala

```diff
@@ -1450,11 +1450,22 @@ object Weeder {
           val head = WeededAst.Predicate.Head.Atom(pred, den, selects, loc)
 
           // The body of the pseudo-rule.
-          val body = where match {
-            case Nil => from
-            case g :: Nil => WeededAst.Predicate.Body.Guard(g, loc) :: from
+          val bodyGuard = where match {
+            case Nil => None
+            case g :: Nil => Some(WeededAst.Predicate.Body.Guard(g, loc))
             case _ => throw InternalCompilerException("Impossible. The list must have 0 or 1 elements.")
           }
+          // Insert fix to all lattice atoms (could limit this to those referenced in the result)
+          val fixedFrom = from.map{
+            case lat @ WeededAst.Predicate.Body.Atom(_, Denotation.Latticenal, _, Fixity.Loose, _, _) =>
+              lat.copy(fixity = Fixity.Fixed)
+            case b => b
+          }
+          // Concat the complete body
+          val body = bodyGuard match {
+            case Some(g) => g :: fixedFrom
+            case None => fixedFrom
+          }
 
```

Comment:
I want to compute `body` in one go without too much fuss. I propose this structure:

```
val guard = where.map(g => WeededAst.Predicate.Body.Guard(g, loc))

val body = guard :: from.map {
  case WeededAst.Predicate.Body.Atom(... write out everything .. we dont use @ ) => 
    Atom(...)
  case pred => pred
}
```
```